src/services/user.service.ts

```diff
@@ -18,47 +59,104 @@ class UserService implements UserServiceInterface {
         this.userModel = userModel;
     }
 
-    async login(login: User['login'], password: User['password']): Promise<Model<User> | null> {
+    async login(
+        login: User['login'],
+        password: User['password']
+    ): Promise<UserAccessToken> {
         const user: Model<User> | null = await this.userModel.findOne(
-            userLogin(login, password)
+            findUser({ login })
         );
 
-        if (!user) return null;
+        const isValidPass = await comparePassword(
+            password,
+            user?.getDataValue('password')
+        );
 
-        return user;
+        if (!user || !isValidPass) return { info: CREDENTIALS_INCORRECT };
+
+        const token = jwt.sign(
+            {
+                data: user.get('id'),
+                exp: Math.floor(Date.now() / 1000) + 60 * 60
```

Comment:
This looks like a good candidate to be converted into a function and stored at auth middleware level instead of service level.

Reply:
Sure, I've created this as `signToken.auth.ts` -  I'm passing the user information in their respective controller to this middleware. 
 
Hope this is okay? Might have missed the mark here when you mentioned converting to a function. If preferred I can just change this into a standard function accepting the userId, rather than (req, res, next) etc.

Reply:
That addresses the comment completely!